projects/angle/index.js

```diff
@@ -84,6 +84,49 @@ async function tvl(chain, block) {
     assets.forEach(({ output }, i) =>
       sdk.util.sumSingleBalance(balances, tokens[i].output, output)
     );
+
+    // AMOs
+    const EUROC = "0x1aBaEA1f7C830bD89Acc67eC4af516284b1bC33c";
+    const curvePool = "0xBa3436Fd341F2C8A928452Db3C5A3670d1d5Cc73";
+    const sdagEUREUROC = "0x63f222079608EEc2DDC7a9acdCD9344a21428Ce7";
+    const cvxagEUREUROCstaker = "0xA91fccC1ec9d4A2271B7A86a7509Ca05057C1A98";
+    const AngleagEUREUROCStaker = "0xC1e8Dba1cbF29f1CaA8343CAe96d5AdFD9bca736";
+    // pool TVL
+    let agEURTVL = await sdk.api.erc20.balanceOf({
+      owner: curvePool,
+      target: agEUR.contract,
+      block: block,
+    });
+    let EUROCTVL = await sdk.api.erc20.balanceOf({
+      owner: curvePool,
+      target: EUROC,
+      block: block,
+    });
+    const totPoolTokenSupply = await sdk.api.erc20.totalSupply({
+      target: curvePool,
+      block: block,
+    });
+    // Angle holdings of Curve agEUREUROC LP tokens (staked on Stake DAO and Convex)
+    let sdagEUREUROCTVL = await sdk.api.erc20.balanceOf({
+      owner: AngleagEUREUROCStaker,
+      target: sdagEUREUROC,
+      block: block,
+    });
+    let cvxagEUREUROCstakerTVL = await sdk.api.erc20.balanceOf({
+      owner: AngleagEUREUROCStaker,
+      target: cvxagEUREUROCstaker,
+      block: block,
+    });
+    const AnglePoolTokenSupply =
+      Number(sdagEUREUROCTVL.output) + Number(cvxagEUREUROCstakerTVL.output);
+    let AngleagEURTVL =
+      (AnglePoolTokenSupply / totPoolTokenSupply.output) * agEURTVL.output;
+    let AngleEUROCTVL =
+      (AnglePoolTokenSupply / totPoolTokenSupply.output) * EUROCTVL.output;
+    console.log(AngleagEURTVL, AngleEUROCTVL);
+
+    sdk.util.sumSingleBalance(balances, agEUR.contract, AngleagEURTVL);
```

Comment:
also, can you remove this part? fairly sure that we are not counting the FRAX part of curve LP in their tvl adapter.

The reason is, we are already counting the tokens backing agEUR

Reply:
I checked beforehand, and FRAX does count the FRAX it holds through AMOs as TVL. They discount them by a CR factor (which is done from the contracts, making it easy to query the results). 
See the [collatDollarBalance() function](https://etherscan.io/address/0x49ee75278820f409ecd67063D8D717B38d66bd71#readProxyContract#F6) on the FRAX3CRV AMO contract. 

We actually had the debate internally yesterday. Counting this agEUR from LP tokens as TVL or not can be seen from two perspectives: 

1. The LP tokens held by the protocol are the assets (TVL) balancing the liabilities (agEUR minted by AMOs) on this pool. The first perspective is that if these agEUR are accounted for in the totalSupply, then they show as liabilities for the protocol. In this case, we need to properly value the LP tokens (agEUR+EUROC) to not have agEUR supply overshooting TVL. 
2. The second perspective would be to not account these minted agEUR in the totalSupply, and not count the LP tokens as TVL. This would actually be "more honest", but unfortunately there is no way to control an ERC20 totalSupply. 

We could definitely add a disclaimer about this in the comments. You could even add one for FRAX, it took some time for me before wrapping my head around this tbh.

Reply:
Did you get the chance to think about it @g1nt0ki ? 
Feel free to dm me on Discord btw (`@tuta#3689`)

Reply:
we'll look into FRAX and remove the unbacked tokens from TVL

Reply:
Ok. Just to be clear, the tokens are **unbacked when minted** by the protocol and deposited in the pool, but are **backed 1 to 1 with the paired token when they are released in the market**, as the counterparty deposits USDC to get FRAX, or EUROC to get agEUR.

Reply:
yeah, i just don't think that we should include minted AMO tokens into TVL in any case

if we do a protocol can just mint infinite, deposit that anywhere and increase their TVL with nothing backing it